packages/shared/src/hooks/analytics/useAnalyticsSharedProps.ts

```diff
@@ -6,11 +6,25 @@ import {
   useState,
 } from 'react';
 import { useRouter } from 'next/router';
+import { get as getCache, set as setCache } from 'idb-keyval';
+import { v4 as uuidv4 } from 'uuid';
 import { AnalyticsEvent } from './useAnalyticsQueue';
 import FeaturesContext from '../../contexts/FeaturesContext';
 import SettingsContext from '../../contexts/SettingsContext';
 import AuthContext from '../../contexts/AuthContext';
 
+const DEVICE_ID_KEY = 'device_id';
+
+const getOrGenerateDeviceId = async (): Promise<string> => {
+  const deviceId = await getCache<string | undefined>(DEVICE_ID_KEY);
```

Comment:
Could you not leverage the usePersistentContext hook?
that technically uses the idb cache

Reply:
The reason I didn't use our hook is that it doesn't support lazy evaluating the default value and I didn't want to introduce it just for this case on one hand, and on the other hand, I didn't want to generate a new UUID when it's not necessary because the device id is already presented. What do you think?